in---
title: "Analysis Report 1: Influence of Physiological factors on female and male microbiome composition
author: "Sanam Sajjadi"
date: "November 04, 2018"
output: github_document
bibliography: references.bib
csl: bioinformatics.csl
---

# Introduction

Skin is the external interface of the human body with the environment and also the home to many non-pathogenic microbiomes. Interestingly estimation shows that the human body is inhabited by ten times as many non-human cells as human cells! 
Recent investigations demonstrate that knowledge of new generation sequencing has become essential for basic biological research specifically, in understanding microbiome diversity on the human skin and gut microbiome. Therefore, human microbiome and new generation sequencing have a critical relationship that plays valuable role in aiding further development of many applications such as forensic studies and medicine. 
There are several studies which demonstrate effective factors in changing microbiome composition. According to these studies environmental factors, host species and diet are important elements causing the huge diversity on human microbiome.An analysis of skin bacterial microbiota through sequencing of 16S ribosomal RNA genes highlighted the extensive diversity of organisms that live on the human skin [@grice2011skin].A vast majority of these colonizing microbiota are beneficial to the human host, while others cause skin disorders such as acne, atopic dermatitis, seborrheic dermatitis, and chronic wounds [@grice2011skin].Changes to microbial communities can contribute to cutaneous disorders, while the maintenance of these communities has a reciprocal nature in order to modulate skin immunity [@sanmiguel2015interactions].
The study of the total number of bacteria that live on the human host provide valuable insight into normal immune system function and the etiology of infectious diseases that were once more elusive [@cox2013sequencing]
Additional research revealed that the skin surface plays a critical role in influencing human health based on the microbial community that lives on it. The epithelial surface is protected below the follicular opening by hair follicles and sweat ducts which increase the skins surface area from 2 m2 to 25m2. Thus, the skin offers a great area for microbial organisms to live [@gallo2017human].
The physical location that a person lives plays a critical role in the type of microbial communities that live on the skin. People living in urban environments have a significantly greater abundance of Trabulsiella, and the overall skin-associated bacterial communities and structure were evident based on location [@ying2015influence]. Further, the study of the microbiology of indoor spaces revealed that humans share their interior environment with countless bacteria and fungi, which may negatively impact the rates of acute and chronic illness [@kelley2013studying]. Understanding the prevalence of bacteria species in different environments allows for considerations related to health and ways to improve these locations.   
New generation sequencing technology is an important analytical tool in forensic studies that allows for application in the DNA database construction, body fluid and species identification, ancestry and phenotypic inference, twin studies, and microbiological analysis [@yang2014application]. In the other related study researhcher asserted that new generation sequencing allows for the analysis of a microbial community to create a more accurate forensic profile. By studying the microbiome that is present in the trace evidence, new generation sequencing helps forensics experts make critical connections that will help criminal justice professionals solve crime [@kuiper2016microbial] . In a study of microbiomes on cell phones and shoes, scientists determined that there was a distinct correlation between microbial community sources which highlights the way in which people interact with their environment [@lax2015forensic]. Additional research found that cell phones contain a significant amount of the personal microbiome of their owners. Human’s DNA extraction and sequencing enabled us to determine gender of the fingertip’s owner in forensic cases. Now microbiome composition can also be used to distinguish gender in forensic cases [@meadow2014mobile]. Also in one of the related studies scientists claimed that sex differences have clear impact on shaping gut microbial composition in mice. Regarding their hypotheses it is possible to differentiate mice genders according to their microbiome composition [@org2016sex] . Through the study of human skin’s microbiome and conducting new generation sequencing techniques Fierer eta al., argued that each individual harbors personally unique, stable and transferable skin microbiome that can be used as “fingerprints” for forensic identification.
The hypotheses described above represent a valuable alternative to the other standard techniques like human DNA analysis and finger print analysis. For instance, it is usually difficult to acquire sufficient human DNA for forensic identification, since sufficient amount of saliva, blood or tissues is needed, however it may be easier to recover bacterial DNA from the touched surfaces. (additional studies needed to confirm this hypotheses) [@fierer2010forensic]. In this analysis report on Fierer’s research I will discuss and demonstrate that there would be a distinct microbial composition differences between genders. I will assume that Physiological differences can be one of the major elements of these differences. I will compare the microbial community on males and females and investigate the factors that may cause these comparisons.

# Methods

## Sample origin and sequencing
Sample collection:  For the keyboard study Fierer and his team had swabbed (25 to 30) keys of three different personal computers and the skin on the front surface of the furthest joint of each fingertip of the owners. Note that those three individuals were healthy and were between 20 and 25 years old at the time of sampling. Two of these participants shared the exact office space. Key board and fingertips were swabbed within 10 minutes of one another, however keyboards had not been touched for more than 30 minutes before swabbing. To compare the microbial communities on these keyboards to other keyboards, Fierer and his team swabbed Space bar keys from 15 different computer keyboards located on the University of Colorado campus. Fierer et al used autoclaved cotton-tipped swabs that were pre-moistened with a sterile solution to collect the swabbing samples. All the swabs were stored at -80 Degree Celsius for less than one week and after that sent for DNA extraction.
DNA Extraction and pyrosequencing: The MO BIO PowerSoil DNA Isolation kit was used to extract genomic DNA by breaking of the frozen cotton tips into bead tubes that contain solution C1. The tubes were incubated for 10 min at 65 Degree Celsius and then shaken using MO BIO vortex adapter. Fierer and his team amplified 16SrRNA genes by utilizing Primer that is best for phylogenetic analysis of pyrosequencing reads.
The researchers conducted PCR reactions and finally replicate amplicons were visualized on 0.1% agarose gel. They cleaned the amplicons by using UltraClean-htp 96-well PCR Clean-up kit or MO BIO.
Fierer et al measured the DNA concertation by Quant-it PicoGreen ds DNA reagent and kit, then after quantitation, cleaned amplicons were combined in equimolar ratios into a single tube.  The Final pool of DNA was separated from the solution on ice after adding NaCl and ethanol. The precipitated DNA was centrifuged multiple times and then supernatant was removed. The Environmental Genomics Core Facility at the University of South Carolina carried out the pyrosequencing on a 454 Life Sciences Genome Sequencer FLX instrument. 

## Computational

These are the methods you used. Should probably be at least a half of a page. At a very minimum should include citations for DADA2 [@callahan2016] and phyloseq [@mcmurdie2013]. Note that these don't count towards the five references you need to cite in the introduction.

# Results

In addition to a minimum of 3-4 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion.

```{r load-libraries, message = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variant
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")
```

```{r extract-sample-and-file-names}
# NOTE: Much of the following follows the DADA2 tutorials available here:
# https://benjjneb.github.io/dada2/tutorial.html
# Accessed October 19, 2017

# set the base path for our input data files
path <- "data/raw_data"

# Sort ensures samples are in order
filenames_forward_reads <- sort(list.files(path, pattern = ".fastq"))

# Extract sample names, assuming filenames have format: SAMPLENAME.fastq
sample_names <- sapply(strsplit(filenames_forward_reads, "\\."), `[`, 1)

# Specify the full path to each of the filenames_forward_reads
filenames_forward_reads <- file.path(path, filenames_forward_reads)
```

```{r check-quality-plots}
# Plots the quality profiles of all the first twenty samples
plotQualityProfile(filenames_forward_reads[1:20])
```

We can see from the quality profiles that most reads tend to get pretty bad in quality after around 200 bases. 

```{r filter-reads}
# Place filtered files in filtered/ subdirectory
# note this will fail if the directory doesn't exist
filter_path <- file.path("output", "filtered")
filtered_reads_path <- file.path(filter_path,
                                 paste0(sample_names,
                                        "_filt.fastq.gz"))

# See ?filterAndTrim for details on the parameters
# See here for adjustments for 454 data:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
filtered_output <- filterAndTrim(fwd = filenames_forward_reads,
                                 filt = filtered_reads_path,
                                 maxLen = 300,
                                 maxN = 0, # discard any seqs with Ns
                                 maxEE = 3, # allow w/ up to 3 expected errors
                                 truncQ = 2, # cut off if quality gets this low
                                 rm.phix = TRUE,
                                 compress = TRUE,
                                 multithread = FALSE)
```

```{r filtered-read-counts-table}
# produce nicely-formatted markdown table of read counts
# before/after trimming
kable(filtered_output,
      col.names = c("Reads In",
                    "Reads Out"))
```

```{r learn-errors}
# this build error models from each of the samples
errors_forward_reads <- learnErrors(filtered_reads_path,
                                    multithread = FALSE)
```

```{r visualize-errors-with-plots}
# quick check to see if error models match data
# (black lines match black points) and are generally decresing left to right
plotErrors(errors_forward_reads,
           nominalQ = TRUE)
```

```{r dereplicate-sequences}
# get rid of any duplicated sequences
dereplicated_forward_reads <- derepFastq(filtered_reads_path,
                                         verbose = FALSE)

# Name the derep-class objects by the sample names
names(dereplicated_forward_reads) <- sample_names
```

```{r run-dada}
# parameters adjusted based on recommendations for 454 data here:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
dada_forward_reads <- dada(dereplicated_forward_reads,
                           err = errors_forward_reads,
                           HOMOPOLYMER_GAP_PENALTY = -1, # reduce penalty bc 454
                           BAND_SIZE = 32) # performs local alignments bc indels

# check dada results
dada_forward_reads
```

```{r make-sequence-table}
# produce the 'site by species matrix'
sequence_table <- makeSequenceTable(dada_forward_reads)
```

The output table has `r nrow(sequence_table)` rows (samples) and `r ncol(sequence_table)` columns (sequence variants). Notice how we can embed R code directly in our markdown text.

```{r histogram-of-sequence-lengths}
# Quick check to look at distribution of trimmed and denoised sequences
hist(nchar(getSequences(sequence_table)),
     main = "Histogram of final sequence variant lengths",
     xlab = "Sequence length in bp")
```

```{r remove-chimeras}
# Check for and remove chimeras
sequence_table_nochim <- removeBimeraDenovo(sequence_table,
                                            method = "consensus",
                                            multithread = FALSE,
                                            verbose = TRUE)

# What percent of our reads are non-chimeric?
non_chimeric_reads <- round(sum(sequence_table_nochim) / sum(sequence_table),
                            digits = 4) * 100
```

After removing chimeras, we were left with `r non_chimeric_reads`% of our cleaned reads.

```{r table-of-pipeline-read-counts}
# Build a table showing how many sequences remain at each step of the pipeline
get_n <- function(x) sum(getUniques(x)) # make a quick function
track <- cbind(filtered_output, # already has 2 columns
               sapply(dada_forward_reads, get_n),
               rowSums(sequence_table),
               rowSums(sequence_table_nochim))

# add nice meaningful column names
colnames(track) <- c("Input",
                     "Filtered",
                     "Denoised",
                     "Sequence Table",
                     "Non-chimeric")

# set the proper rownames
rownames(track) <- sample_names

# produce nice markdown table of progress through the pipeline
kable(track)
```

```{r assign-taxonomy}
# assigns taxonomy to each sequence variant based on a supplied training set
# made up of known sequences
taxa <- assignTaxonomy(sequence_table_nochim,
                       "data/training/rdp_train_set_16.fa.gz",
                       multithread = FALSE,
                       tryRC = TRUE) # also check with seq reverse compliments

# show the results of the taxonomy assignment
unname(taxa); #
```

```{r extract-sequences-to-fasta}
# we want to export the cleaned, trimmed, filtered, denoised sequence variants
# so that we can build a phylogeny - we'll build the phylogeny outside of R
# but we need the fasta file to do so. We keep the names of each sequence as the
# sequence itself (which is rather confusing), because that's how DADA2 labels
# it's columns (e.g. 'species')
# function taken from https://github.com/benjjneb/dada2/issues/88
export_taxa_table_and_seqs <- function(sequence_table_nochim,
                                       file_seqtab,
                                       file_seqs) {
  seqtab_t <- as.data.frame(t(sequence_table_nochim)) # transpose to data frame
  seqs <- row.names(seqtab_t) # extract rownames
  row.names(seqtab_t) <- seqs # set rownames to sequences
  outlist <- list(data_loaded = seqtab_t)
  mctoolsr::export_taxa_table(outlist, file_seqtab) # write out an OTU table
  seqs <- as.list(seqs)
  seqinr::write.fasta(seqs, row.names(seqtab_t), file_seqs) # write out fasta
}

# actually run the function, with the names of the files we want it to create
# and where to put them
export_taxa_table_and_seqs(sequence_table_nochim,
                           "output/sequence_variants_table.txt",
                           "output/sequence_variants_seqs.fa")
```


```{r read-in-metadata-and-create-phyloseq}
# Next we want to read in the metadata file so we can add that in too
# This is not a csv file, so we have to use a slightly different syntax
# here the `sep = "\t"` tells the function that the data are tab-delimited
# and the `stringsAsFactors = FALSE` tells it not to assume that things are
# categorical variables
metadata_in <- read.table(paste0("data/metadata/",
                    "fierer_hand_bacteria_SRA_study_ERP022626_SraRunTable.txt"),
                          sep = "\t",
                          header = TRUE,
                          stringsAsFactors = FALSE,
                          row.names = 6) # sets sample IDs to row names

# read in the phylogeny, which was created from the fasta exported above
# in Geneious by aligning the sequences with MAFFT and then building a
# Maximum-Likelihood tree with RAxML
tree_in <- read_tree("output/sequence_variants_MAFFT_FastTree.newick")

# Construct phyloseq object (straightforward from dada2 outputs)
phyloseq_obj <- phyloseq(otu_table(sequence_table_nochim,
                                   taxa_are_rows = FALSE), # sample-spp matrix
                         sample_data(metadata_in), # metadata for each sample
                         tax_table(taxa), # taxonomy for each sequence variant
                         phy_tree(tree_in)) # phylogeny from sequence variants
```


```{r histogram-showoing-the-abundance-of-sequences-in-both-genders}
# bar plot of samples
plot_bar(phyloseq_obj, x = "Sex")
```


**Figure 1**: Histogram that describes the abundance of sequences in both females and males.

 Most of the sequences were from males and only few females were in this project.


```{r table-contains-sources-of-sequences-in-both-genders}
table(factor(metadata_in$sample_source), factor(metadata_in$Sex))

```

**Figure 2**: Table that describes the sources of the sequences in both genders.

Most of the sequences were from males samples and only 3 females samples were 
swabbed from spacebar key. 



```{r histogram-examining-diversity-of-phyla-acroos-sample-set-in-both-gender}
sex_phyloseq_obj <- subset_samples(phyloseq_obj, sample_source = "sex")
plot_bar(sex_phyloseq_obj,
         x = "Phylum",
         fill = "Class")


```

**Figure3** Diversity of different phyla in both gender colored by class.

This graph shows that most of the bacteria were from Actinobactria and Firmicutes Phyla.



```{r histogram-showing-the-abundance-of-different-bacteria-in-females-samples}
female_phyloseq_obj <- subset_samples(phyloseq_obj, Sex == "female")
plot_bar(female_phyloseq_obj, x = "Class")
```


**Figure4**: Histogram that represents the diversity of bacteria in female samples.

This histogram shows that most of the bacteria were from Actinobacteria,
Bacilli and Chloroplast classes.
       
       
       
```{r histogram-showing-the-abundance-of-different-bacteria-in-males-samples}
male_phyloseq_obj <- subset_samples(phyloseq_obj, Sex == "male")
plot_bar(male_phyloseq_obj, x = "Class")
```
       
**Figure 5**: Histogram that represents the diversity of bacteria in male samples.

This histogram shows that most of the bacteria were from Actinobacteria and Bacillii classes.




# Discussion

Add around 2-3 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited


